File: common/src/main/java/net/abitmorehex/abitmorehex.java
package net.abitmorehex;

import net.abitmorehex.registry.abitmorehexBlockRegistry;
import net.abitmorehex.registry.abitmorehexIotaTypeRegistry;
import net.abitmorehex.registry.abitmorehexItemRegistry;
import net.abitmorehex.registry.abitmorehexActionRegistry;
import net.minecraft.util.Identifier;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class abitmorehex {
    public static final String MOD_ID = "abitmorehex";
    public static final Logger LOGGER = LogManager.getLogger(MOD_ID);

    public static void init() {
        LOGGER.info("aBitMore hexes is aBit and Loaded!");

        // If you have other registries, init them as well
        abitmorehexItemRegistry.init();
        abitmorehexBlockRegistry.init();
        abitmorehexIotaTypeRegistry.init();

        // This is the important part: register your actions/patterns
        abitmorehexActionRegistry.init();

        LOGGER.info("Config path: " + DummyAbstractions.getConfigDirectory()
                .toAbsolutePath().normalize());
    }

    public static Identifier id(String path) {
        return new Identifier(MOD_ID, path);
    }
}

File: Common/src/main/kotlin/net/abit/abitmorehex/Abitmorehex.kt
package net.abit.abitmorehex

import net.minecraft.resources.ResourceLocation
import org.apache.logging.log4j.LogManager
import org.apache.logging.log4j.Logger
import net.abit.abitmorehex.config.AbitmorehexConfig
import net.abit.abitmorehex.misc.LootInjector
import net.abit.abitmorehex.networking.AbitmorehexNetworking
import net.abit.abitmorehex.recipes.AbitmorehexRecipesAdditions
import net.abit.abitmorehex.registry.*
import net.abit.abitmorehex.registry.AbitmorehexActions

object Abitmorehex {

    const val MODID = "abitmorehex"

    @JvmField
    val LOGGER: Logger = LogManager.getLogger(MODID)

    @JvmStatic
    fun id(path: String) = ResourceLocation(MODID, path)

    fun init() {
        AbitmorehexConfig.init()
        initRegistries(
            AbitmorehexIotaTypes,
            AbitmorehexActions,
            AbitmorehexEffects,
            AbitmorehexBlocks,
            AbitmoreItems,
            AbitmorehexBlockEntities,
            AbitmorehexCreativeTabRegistries,
            AbitmorehexRecipeTypes,
            AbitmorehexRecipes
        )
        LootInjector.init()
        AbitmorehexNetworking.init()
    }
}
